R/volcanoPlot.R

```diff
@@ -114,244 +108,381 @@ volcanoPlot <- function(contrastDF,
+        warning("ylab must be a singular value of class character. Assigning default value 'NULL'.")
+        ylab <- NULL
     }
 
-    if (is.null(title)) {
-        title <- ""
+    if (any(is.null(symbolSize),
+            !is.numeric(symbolSize),
+            length(symbolSize)  != 3,
+            length(unique(symbolSize)) < 2,
+            !all(symbolSize >= 0))) {
+        warning("symbolSize must be a vector of 3 integer values, at least 2 of them are different. Assigning default values 10, 4, 10.")
+        symbolSize  <-  c(10, 4, 10)
+
     }
 
-    x <- make.names(colnames(contrastDF)[colnames(contrastDF) == logRatioCol])
-    colnames(contrastDF)[colnames(contrastDF) == logRatioCol] <- make.names(colnames(contrastDF)[colnames(contrastDF) == logRatioCol])
-    # Add NegativeLogP column
-    contrastDF$NegativeLogP <- -1*log10(contrastDF[,pvalCol])
-    y <- "NegativeLogP"
-
-    contrastDF$group <- NA
-    contrastDF$group <- ifelse(contrastDF[[pvalCol]] <= pthreshold,
-                               ifelse(contrastDF[[logRatioCol]] >= 0, "Increased", "Decreased"),
-                               "No Change")
-    contrastDF$group <- contrastDF$group %>%
-        factor(levels = c("Increased", "Decreased", "No Change"))
-
-    # plotType
-    if (plotType == "canvasXpress") {
-        symbolFill <- sapply(symbolFill, rgbaConversion, alpha = alpha, USE.NAMES = FALSE)
-
-        ## Create the canvasXpress cx.data and var.annot
-        cx.data <- data.frame(a = contrastDF[colnames(contrastDF) == x],
-                              b = contrastDF[colnames(contrastDF) == y])
-        colnames(cx.data) <- c(x, y)
-        var.annot <- data.frame(Group = contrastDF$group, LogInt = contrastDF$LogInt)
-        rownames(var.annot) <- rownames(cx.data)
-        events <- NULL
-
-        # Add geneSym labels in tooltips
-        if (!missing(geneSymCol)) {
-            var.annot <- cbind(var.annot, GeneLabel = contrastDF[[geneSymCol]])
-            events <- htmlwidgets::JS("{ 'mousemove' : function(o, e, t) {
-                                                if (o != null && o != false) {
-                                                    if (o.objectType == null) {
-                                                        t.showInfoSpan(e, '<b>' + o.y.vars + '</b> <br/>' +
-                                                        '<b>' + 'GeneLabel'  + '</b>' + ': ' + o.z.GeneLabel[0] + '<br/>' +
-                                                        '<b>' + o.y.smps[0]  + '</b>' + ': ' + o.y.data[0][0] + '<br/>' +
-                                                        '<b>' + o.y.smps[1]  + '</b>' + ': ' + o.y.data[0][1]);
-                                                    } else {
-                                                        t.showInfoSpan(e, o.display);
-                                                    };
-                                                }; }}")
-        }
+    if (any(is.null(symbolShape),
```

Comment:
Can you check this criteria? It doesnt look right. This will always  be FALSE because plotType can only either be canvasXpress or ggplot.